tests/www/geiq_views/tests.py

```diff
@@ -0,0 +1,237 @@
+        date_debut="2022-12-15:00:00+01:00",
+        date_fin="2023-01-10:00:00+01:00",
+    )
+
+    def _fake_get_all_contracts(self, geiq_id):
+        assert geiq_id == assessment.label_id
+        return [dict(contract), dict(contract_with_prequal)]
+
+    def _fake_get_all_prequalifications(self, geiq_id):
+        assert geiq_id == assessment.label_id
+        return [prequal_of_contract]
+
+    mocker.patch.object(geiq_label.LabelApiClient, "get_all_contracts", _fake_get_all_contracts)
+    mocker.patch.object(geiq_label.LabelApiClient, "get_all_prequalifications", _fake_get_all_prequalifications)
+
+    response = client.post(reverse("geiq:label_sync", kwargs={"assessment_pk": assessment.pk}))
+    assessment.refresh_from_db()
+    assert assessment.last_synced_at is not None
+    assertContains(response, "Dernière synchronisation:")
+
+    response = client.post(
+        assessment_info_url,
+        data={"activity_report_file": pdf_file, "up_to_date_information": True},
+    )
+    assertContains(
+        response,
+        '<span class="badge badge-sm rounded-pill text-nowrap bg-info">Bilan à l’étude</span>',
+    )
+    assessment.refresh_from_db()
+    assert assessment.submitted_at is not None
+    assert assessment.submitted_by == geiq_user
+    pdf_file.seek(0)
+    with default_storage.open(assessment.activity_report_file_id) as saved_report:
+        assert saved_report.read() == pdf_file.read()
+
+    institution_membership = InstitutionMembershipFactory(institution__kind=InstitutionKind.DDETS_GEIQ)
+    assessment.reviewed_at = timezone.now()
+    assessment.reviewed_by = institution_membership.user
+    assessment.review_institution = institution_membership.institution
+    assessment.review_state = ReviewState.ACCEPTED
+    assessment.review_comment = "Bon boulot !"
+    assessment.save()
+
+    response = client.get(assessment_info_url)
+    assertContains(
+        response,
+        (
+            '<span class="badge badge-sm rounded-pill text-nowrap bg-success">'
+            "Financement : totalité de l’aide accordée</span>"
+        ),
+    )
+
+
+def test_state_snapshot(client, snapshot):
+    geiq_membership = CompanyMembershipFactory(company__kind=CompanyKind.GEIQ, company__for_snapshot=True)
+    assessment = ImplementationAssessmentFactory(campaign__year=2023, company=geiq_membership.company)
+    geiq_user = geiq_membership.user
+    institution_membership = InstitutionMembershipFactory(
+        user__last_name="GADGET",
+        user__first_name="Inspecteur",
+        institution__kind=InstitutionKind.DDETS_GEIQ,
+        institution__department=75,
+        institution__name="DDETS 75",
+    )
+    ddets = institution_membership.institution
+    ddets_user = institution_membership.user
+    assessment_info_url = reverse("geiq:assessment_info", kwargs={"assessment_pk": assessment.pk})
+
+    def check_snapshots(user, name):
```

Comment:
Pour clarifier un peu.
```suggestion
    def check_snapshots(user, snapshot_name):
```